### PR TITLE
set up authorization

### PR DESCRIPTION
**Pull request by betaredex**

### CODE REVIEW

File: bin/pman [hunk 1]
```diff
@@ -155,6 +155,20 @@ parser.add_argument(
...
 )
+parser.add_argument(
+    '--enableTokenAuth',
+    action  = 'store_true',
+    help    = 'Enables token based authorization and can be configured to look for a .ini file or an openshift secret',
+    dest    = 'b_tokenAuth',
+    default = False
+)
+parser.add_argument(
+    '--tokenPath',
```
Comment (danmcp) [inline bin/pman]: A couple of things we talked about for this path was having the value obfuscated and allowing for multiple keys so they could be rotated easier.

File: pman/Auth.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
+# A naive Class that implements basic token based auth
```
Comment (danmcp) [inline pman/Auth.py]: This looks like a lot of copied code.  Can we move to a shared library?

Comment (danmcp) [reply pman/Auth.py]: Maybe pfmisc

File: pman/pman.py [hunk 3]
```diff
@@ -533,7 +540,7 @@ def __init__(self, **kwargs):
         self.str_debugFile      = '/dev/null'
         self.b_debugToFile      = True
         self.pp                 = pprint.PrettyPrinter(indent=4)
-
+
```
Comment (danmcp) [inline pman/pman.py]: Why?

Comment (betaredex) [reply pman/pman.py]: That must have been a mistake, I'll fix it.

File: pman/Auth.py (hunk 2 above)
Comment (betaredex) [reply pman/Auth.py]: I thought about that but didn't know what our library structure was like. I'll look into that (and make the necessary changes in pfioh as well)

File: bin/pman (hunk 1 above)
Comment (betaredex) [reply bin/pman]: Not certain what you mean by that. The way its set up now is that all the keys are in a single config file which is encrypted with an openshift secret. How exactly do you want me to set it up?

**Pull request closed without merging**